client/src/main/java/io/pravega/client/stream/EventPointer.java

```diff
@@ -29,6 +30,13 @@
      * @return Implementation of EventPointer interface
      */
     EventPointerInternal asImpl();
+
+    /**
+     * This can be Used Externally.
+     *
+     * @return Implementation of EventPointer interface
+     */
+    EventPointerExternal asExternalImpl();
```

Comment:
I don't see the how this is different from what `asImpl()` is doing? 
My objection to making the methods public on `EventPointerInternal` was that they were public, not that they should be public but were on class called Internal. Please find a way to do this without making the methods public.